src/app/util/attribute-table.cpp

```diff
@@ -46,33 +46,13 @@ using namespace chip;
 EmberAfStatus emberAfWriteAttributeExternal(EndpointId endpoint, ClusterId cluster, AttributeId attributeID, uint8_t * dataPtr,
                                             EmberAfAttributeType dataType)
 {
-    EmberAfAttributeWritePermission extWritePermission =
-        emberAfAllowNetworkWriteAttributeCallback(endpoint, cluster, attributeID, dataPtr, dataType);
-    switch (extWritePermission)
-    {
-    case EmberAfAttributeWritePermission::DenyWrite:
-        return EMBER_ZCL_STATUS_FAILURE;
-    case EmberAfAttributeWritePermission::AllowWriteNormal:
-    case EmberAfAttributeWritePermission::AllowWriteOfReadOnly:
-        return emAfWriteAttribute(endpoint, cluster, attributeID, dataPtr, dataType,
-                                  (extWritePermission == EmberAfAttributeWritePermission::AllowWriteOfReadOnly), false);
-    default:
-        return (EmberAfStatus) extWritePermission;
-    }
+    return emAfWriteAttribute(endpoint, cluster, attributeID, dataPtr, dataType, false /* override read-only */);
 }
 
 EmberAfStatus emberAfWriteAttribute(EndpointId endpoint, ClusterId cluster, AttributeId attributeID, uint8_t * dataPtr,
                                     EmberAfAttributeType dataType)
 {
-    return emAfWriteAttribute(endpoint, cluster, attributeID, dataPtr, dataType,
-                              true,   // override read-only?
-                              false); // just test?
-}
-
-EmberAfStatus emberAfReadAttribute(EndpointId endpoint, ClusterId cluster, AttributeId attributeID, uint8_t * dataPtr,
-                                   uint16_t readLength)
-{
-    return emAfReadAttribute(endpoint, cluster, attributeID, dataPtr, readLength, nullptr);
```

Comment:
The "emAf" bits are basically not public API.  The "emberAf" bits are.  That's why there was the forwarding going on there....

Reply:
I was unclear why because one was just calling the other with identical parameters.

If this is a naming convention, I can rename this to emberAfReadAttribute instead of `emAf...`. Are you thinking that is sufficient or do you imagine the 'public version' will eventually do more?

For background of where I am going with these changes: looking to make the API as small as we use and then I would like to place it into something replaceable via an interface, to get testability for clusters without requiring custom zap-regen for every test app.  This is how I got to touching read/write of attributes overall.

Reply:
> If this is a naming convention, I can rename this to emberAfReadAttribute instead of emAf...

And move the declaration into the "more public" af.h header?  That would be fine from my point of view.

Reply:
For what its worth, I hate it when the header name and impl name differ :/ Will not fix here since I plan to add a replaceable interface/shim and  anyway.

Reply:
I will move emberAfWriteAttributeExternal too then.